ex4/VendingDP.Lib/CommandExecutor.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace VendingDP.Lib
+{
+    public class CommandExecutor
+    {
+        private const string HELLO = "Hello! What would you like to order?";
+        private const string UNKNOWN = "Unknown Command";
+        private const string END_EMPTY_ORDER = "Ending an empty order. This has no effect.";
+        public void Introduce()
+        {
+            Console.WriteLine(HELLO);
+            PrintPossibleCommands();
+        }
+        public void Execute(string command, ref Order currentOrder)
```

Comment:
This is a little bit overkill method. You design of commands imply `switch-case` in every functions you deal with the all possible commands. Moreover, if you wand to add a new command there are so many places inside the code you should change. A cleaner design would be to to create an interface with a function called `Execute` that all commands implement, and you just call it.
Another tip - if you method exceeded 20 lines you maybe do more than one thing in that method.